app/src/main/java/com/android/sample/ui/activity/EditActivity.kt

```diff
@@ -229,16 +235,60 @@ fun EditActivityScreen(
                       }
                     }
               }
+          Spacer(modifier = Modifier.height(8.dp))
 
-          OutlinedTextField(
-              value = location,
-              onValueChange = { location = it },
-              label = { Text("Location") },
-              modifier = Modifier.padding(8.dp).fillMaxWidth().testTag("inputLocationEdit"),
-              placeholder = {
-                Text(text = stringResource(id = R.string.request_location_activity))
-              },
-          )
+          // Location Input with dropdown using ExposedDropdownMenuBox
+          ExposedDropdownMenuBox(
+              expanded = showDropdown && locationSuggestions.isNotEmpty(),
+              onExpandedChange = { showDropdown = it } // Toggle dropdown visibility
+              ) {
+                OutlinedTextField(
+                    value = locationQuery,
+                    onValueChange = {
+                      locationViewModel.setQuery(it)
+                      showDropdown = true // Show dropdown when user starts typing
+                    },
+                    label = { Text("Location") },
+                    placeholder = { Text("Enter an Address or Location") },
+                    modifier =
+                        Modifier.menuAnchor() // Anchor the dropdown to this text field
+                            .fillMaxWidth()
+                            .testTag("inputLocationEdit"),
+                    singleLine = true)
+
+                // Dropdown menu for location suggestions
+                // Another approach using DropdownMenu is in EditToDo.kt
+                ExposedDropdownMenu(
+                    expanded = showDropdown && locationSuggestions.isNotEmpty(),
+                    onDismissRequest = { showDropdown = false }) {
+                      locationSuggestions.filterNotNull().take(3).forEach { location ->
+                        DropdownMenuItem(
+                            text = {
+                              Text(
+                                  text =
+                                      location.name.take(30) +
+                                          if (location.name.length > 30) "..."
+                                          else "", // Limit name length
+                                  maxLines = 1 // Ensure name doesn't overflow
+                                  )
+                            },
+                            onClick = {
```

Comment:
Modularise this by creating a function that would be called here and in create Activity. Could be great for next sprint